src/bootstrap/builder.rs

```diff
@@ -530,6 +520,18 @@ impl<'a> Builder<'a> {
         Some(help)
     }
 
+    fn get_builder(build: &Build, kind: Kind, paths: Vec<PathBuf>) -> Builder<'_> {
+        Builder {
+            build,
+            top_stage: build.config.stage.unwrap_or(1),
+            kind,
+            cache: Cache::new(),
+            stack: RefCell::new(Vec::new()),
+            time_spent_on_dependencies: Cell::new(Duration::new(0, 0)),
+            paths,
+        }
+    }
+
```

Comment:
Why not implement `Default` instead of having this?

Reply:
Is there a sane default for Build, Kind, and Paths? I don't want to implement Default if the default value isn't usable on its own.

Reply:
I mean one could use `{ build, kind, paths, ..Default::default() }`.

Reply:
Right, but you can't construct `Default::default()` because there is no `impl Default for Build`.